Pcap++/header/PcapDevice.h

```diff
@@ -18,6 +18,79 @@ namespace pcpp
 	//Forward Declaration - required for IPcapDevice::matchPacketWithFilter
 	class GeneralFilter;
 
+	namespace internal
+	{
+		/// @class PcapHandle
+		/// @brief A wrapper class for pcap_t* which is the libpcap packet capture descriptor.
+		/// This class is used to manage the lifecycle of the pcap_t* object
+		class PcapHandle
+		{
+		public:
+			/// @brief Creates an empty handle.
+			PcapHandle() = default;
+			/// @brief Creates a handle from the provided pcap descriptor.
+			/// @param pcapDescriptor The pcap descriptor to wrap.
+			explicit PcapHandle(pcap_t* pcapDescriptor);
+
+			PcapHandle(const PcapHandle&) = delete;
+			PcapHandle(PcapHandle&& other) noexcept;
+
+			PcapHandle& operator=(const PcapHandle&) = delete;
+			PcapHandle& operator=(PcapHandle&& other) noexcept;
+			PcapHandle& operator=(std::nullptr_t) noexcept;
+
+			~PcapHandle();
+
+			/// @brief Check if the handle is not null.
+			/// @return True if the handle is not null, false otherwise.
+			bool hasValue() const
+			{
+				return m_PcapDescriptor != nullptr;
+			}
+
+			/// @brief Access the underlying pcap descriptor.
+			/// @return The pcap descriptor.
+			pcap_t* value() const
+			{
+				return m_PcapDescriptor;
+			}
+
+			/// @brief Releases ownership of the handle and returns the pcap descriptor.
+			/// @return The pcap descriptor or nullptr if no handle is owned.
+			pcap_t* release();
+
+			/// @brief Helper function to retrieve the last error string for this handle.
+			/// @return The last error string.
+			std::string getLastError() const;
+
+			/// @brief Helper function to retrieve a view of the last error string for this handle.
+			/// @return A view of the last error string.
+			/// @remarks This function is more efficient than getLastError() as it does not copy the string.
+			char const* getLastErrorView() const;
+
+			/// @brief Implicit conversion to bool.
+			/// @return True if the handle is not null, false otherwise.
+			operator bool() const
+			{
+				return hasValue();
+			}
+
+			bool operator==(std::nullptr_t) const
+			{
+				return !hasValue();
+			}
+			bool operator!=(std::nullptr_t) const
+			{
+				return hasValue();
+			}
```

Comment:
do we use this anywhere?

Reply:
Quite a lot of places actually. It is to allow existing comparisons like `if (pcapDescriptor == nullptr)` without too much changes.